extensions/notebook/src/protocol/notebookUriHandler.ts

```diff
@@ -21,27 +21,26 @@ export class NotebookUriHandler implements vscode.UriHandler {
 	constructor() {
 	}
 
-	handleUri(uri: vscode.Uri): vscode.ProviderResult<void> {
-		switch (uri.path) {
+	handleUri(uri: vscode.Uri): Thenable<void> {
+		switch (uri?.path) {
```

Comment:
This isn't typed as being undefined in vscode.d.ts so I would say making this able to handle that scenario is unnecessary.